file/writable_file_writer.h

```diff
@@ -297,7 +304,9 @@ class WritableFileWriter {
   }
   void set_seen_error() { seen_error_.store(true, std::memory_order_relaxed); }
 
-  IOStatus GetWriterHasPreviousErrorStatus() {
+  IOStatus AssertFalseAndGetStatusForPrevError() {
+    // This should only happen if SyncWithoutFlush() was called.
+    assert(sync_without_flush_called_);
```

Comment:
This adds back an arbitrary restriction in a set of circumstances (no `SyncWithoutFlush()`) that feels totally unrelated to the restriction itself. It took a while to remove it despite all that. Can we leave it deleted?

Reply:
Sure - sorry I missed this part of the PR

Reply:
Fixed